obal/data/module_utils/obal.py

```diff
@@ -10,6 +10,23 @@
     from .koji_wrapper import koji, KojiCommandError # pylint:disable=import-error,no-name-in-module
 
 
+def macro_lookup(command, scl=None, dist=None, macros=None):
+    """run a macro lookup command"""
+    if dist:
+        command += ['--define', '"dist %s"' % dist]
+    else:
+        command += ['--undefine', 'dist']
+
+    if scl:
+        command += ['--define', '"scl %s"' % scl]
+
+    if macros is not None:
+        for (macro, value) in macros.items():
+            command += ['--define', '"%s %s"' % (macro, value)]
+
+    return subprocess.check_output(' '.join(command), universal_newlines=True, shell=True)
```

Comment:
I know this is copied code, but I wonder if `shell=True` is really needed here.

Reply:
Without that we get:

```
      File "/tmp/ansible_copr_build_payload_5hf5no4a/ansible_copr_build_payload.zip/ansible/module_utils/obal.py", line 27, in macro_lookup
      File "/usr/lib64/python3.10/subprocess.py", line 421, in check_output
        return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
      File "/usr/lib64/python3.10/subprocess.py", line 503, in run
        with Popen(*popenargs, **kwargs) as process:
      File "/usr/lib64/python3.10/subprocess.py", line 971, in __init__
        self._execute_child(args, executable, preexec_fn, close_fds,
      File "/usr/lib64/python3.10/subprocess.py", line 1847, in _execute_child
        raise child_exception_type(errno_num, err_msg, err_filename)
    FileNotFoundError: [Errno 2] No such file or directory: 'rpmquery --queryformat %{name} --package /tmp/SRPMs/hello-2.10-1.src.rpm --undefine dist'
```

Reply:
Haha. Yeah. Drop that join and pass a list, then it should work.

Reply:
Lets see what the tests say...

Reply:
I think we need to drop the quotes (when constructing the define) now, as there is no shell anymore and thus no quoting is needed

Reply:
That seems to have done it.